Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -46,8 +46,11 @@ TcpClient::TcpClient(TcpClientDataDelegate onReceive)
 
 TcpClient::~TcpClient()
 {
-	delete stream;
-	stream = NULL;
+	if (stream != NULL)
```

Comment:
That change is not needed. `delete` works nicely with pointers that have NULL value.